README.md
ubuntu-additional
=================

[![GitHub license](https://sinfallas.files.wordpress.com/2016/02/gpl.png)](https://github.com/sinfallas/ubuntu-additional/blob/master/LICENSE)
[![DOI](https://zenodo.org/badge/4102/sinfallas/ubuntu-additional.svg)](https://zenodo.org/badge/latestdoi/4102/sinfallas/ubuntu-additional)
![Status](https://api.travis-ci.org/sinfallas/ubuntu-additional.svg) 

Install aditional programs for several version of ubuntu


changelog
commit f3a3f740d83cf6624004cea366dbdfaff3e779a4 (HEAD -> master, origin/master, origin/HEAD)
Author: Jesus <sinfallas@gmail.com>
Date:   Sun Jan 31 16:06:54 2021 -0400

    * changelog

commit df687c27330929f42dfc53f342beb12abd17f0e1
Author: config <Palencia>
Date:   Mon Jul 6 22:04:56 2020 -0400

    * changelog

commit febb2edc3e45f39c07fbea024d71b9ef835fb745
Author: jose perez <sinfallas@gmail.com>
Date:   Mon Apr 27 16:53:26 2020 -0400

    permisos

commit b1b9ed4b392ab28de871844d687c246897634425
Author: jose perez <sinfallas@gmail.com>
Date:   Sun Sep 30 13:28:32 2018 -0400

    * changelog

commit 693f4ce2481aacedf8c32c5643416619f3a1991b (tag: v2.3)
Author: jose perez <sinfallas@gmail.com>
Date:   Mon Feb 29 21:38:37 2016 -0430

    * changelog

commit db4d8756e2256224cda7c15e79c028d18ab8aa85
Author: jose perez <sinfallas@gmail.com>
Date:   Mon Feb 29 21:38:23 2016 -0430

    * logo

commit d809ba00e29f8ff2b29d0d263517910f9d9ce518
Author: jose perez <sinfallas@gmail.com>
Date:   Thu Dec 17 20:29:10 2015 -0430

    * changelog

commit 712ac5ba8a9c82752ac197dc0882de51a5540112
Author: jose perez <sinfallas@gmail.com>
Date:   Sat Dec 12 00:16:43 2015 -0430

    * changelog

commit 6f89d51e4c4c076a8f22900ebeaf6ace5f9b66b0
Author: jose perez <sinfallas@gmail.com>
Date:   Sat Dec 12 00:13:42 2015 -0430

    bugfix

commit 1a873a238916a177eb435215e057349f5cf3cd81
Author: jose perez <sinfallas@gmail.com>
Date:   Thu Dec 10 22:20:28 2015 -0430

    * changelog

commit b3e70b1442475e1b0d6fb82d6a6d1b5e2b8fd484
Author: jose perez <sinfallas@gmail.com>
Date:   Thu Dec 10 22:17:46 2015 -0430

    bugfix

commit 7eaff3af00856c9647924e73011446a2c6b222e5
Author: jose perez <sinfallas@gmail.com>
Date:   Thu Dec 10 21:20:00 2015 -0430

    * changelog

commit 2a738a3a7864d93610aa79216996ea4756e0b326
Author: jose perez <sinfallas@gmail.com>
Date:   Thu Dec 10 21:16:55 2015 -0430

    optimizaciones

commit 4ffabfb6a6fac719be2f9dd2effc96c2fd1774b6 (tag: v2.2)
Author: jose perez <sinfallas@gmail.com>
Date:   Sun Dec 6 18:05:14 2015 -0430

    * changelog

commit 76bea20fadf0d008c90dd0ea7e7ee4fe02d7385b
Author: jose perez <sinfallas@gmail.com>
Date:   Tue Oct 13 21:58:34 2015 -0430

    coveralls

commit 3a1d8040148ec4bc451ff8a15dc2381736ddf8da
Author: jose perez <sinfallas@gmail.com>
Date:   Mon Oct 12 12:45:59 2015 -0430

    licencia

commit 8e9df5033decb4216b134ffaf676df066cd6fd3f
Author: jose perez <sinfallas@gmail.com>
Date:   Mon Aug 10 20:32:25 2015 -0430

    * changelog

commit 083419f161099f5b8bddfe8c6f1c64c6c9338b3e
Author: jose perez <sinfallas@gmail.com>
Date:   Mon Aug 10 20:32:09 2015 -0430

    * doi

commit 84be294102f6de6089c5d300296a507590d87c62 (tag: v2.1)
Author: jose perez <sinfallas@gmail.com>
Date:   Sat Aug 1 21:52:45 2015 -0430

    * changelog

commit c68a78c0a4e1e16475a79d065ca86685dc2ee405
Author: jose perez <sinfallas@gmail.com>
Date:   Sat Aug 1 21:52:24 2015 -0430

    * coveralls

commit 493a5621aecc16b8d25ad7641599e670ff58c4ef
Author: jose perez <sinfallas@gmail.com>
Date:   Thu Apr 2 12:17:28 2015 -0430

    * changelog

commit d9116d9424cb50c14bb828a6019a962610d011e3
Author: jose perez <sinfallas@gmail.com>
Date:   Thu Apr 2 12:16:48 2015 -0430

    * se agrega el archivo fstab

commit 6b16bbfeee2f09e8a8de0c9861ed331ae290cdbc (tag: v2.0)
Author: jose perez <sinfallas@gmail.com>
Date:   Fri Mar 6 17:09:57 2015 -0430

    * changelog

commit 907e1b54a12fde6072825e52dc1f4e7006eaa02d
Author: jose perez <sinfallas@gmail.com>
Date:   Fri Mar 6 17:09:30 2015 -0430

    * se completa la traduccion

commit 8f241edcf77f498ec4cea889b6ccc83996ae3997
Author: jose perez <sinfallas@gmail.com>
Date:   Sun Feb 22 16:33:11 2015 -0430

    * changelog

commit e50b8a5591b15adbe5f38ba5f220285b82461484
Author: jose perez <sinfallas@gmail.com>
Date:   Sun Feb 22 16:33:01 2015 -0430

    * bugfix

commit 7787130fbc7cb19530efaee4850cb18138e48719
Author: jose perez <sinfallas@gmail.com>
Date:   Sat Feb 21 10:50:59 2015 -0430

    * changelog

commit 7236ad22ad33d3aee9d171b4decebeb1b8610dc0
Author: jose perez <sinfallas@gmail.com>
Date:   Sat Feb 21 10:50:13 2015 -0430

    * se elimina el script de virtualizacion porque tiene su propio repo
    * se agrega el archivo para activar lla hibernacion

commit 6eb287a4fd58b81c534c14deffdfab63a01e909b
Author: jose perez <sinfallas@gmail.com>
Date:   Fri Feb 20 20:02:10 2015 -0430

    * bugfix

commit 7375201122acb7955bbf31c2bbca3bea1e8a6d85
Author: sinfallas <sinfallas@gmail.com>
Date:   Mon Feb 2 22:26:10 2015 -0430

    * changelog

commit c66c3dae39ddeb845cd705180d757bd0bb656b7e
Author: sinfallas <sinfallas@gmail.com>
Date:   Mon Feb 2 22:25:47 2015 -0430

    * se corrigen parametros del fstab

commit 7f494c7db573a7216528b9105220bc4422240ab6
Author: sinfallas <sinfallas@gmail.com>
Date:   Thu Jan 29 19:55:45 2015 -0430

    * changelog

commit bb0211ae07a42d5f01b3d3bff94a08ca15b00dba
Author: sinfallas <sinfallas@gmail.com>
Date:   Thu Jan 29 19:55:11 2015 -0430

    * se agrega bchunk

commit cad2b58e12eab4b92e2b7e5232316e020f70427b
Author: sinfallas <sinfallas@gmail.com>
Date:   Thu Jan 8 00:48:29 2015 -0430

    * changelog

commit a9fdc029eaddfebb0a0e01ccf140ceabe826a5a1
Author: sinfallas <sinfallas@gmail.com>
Date:   Thu Jan 8 00:47:30 2015 -0430

    bugfix

commit 2f21bf1d694cc5f290d60ebf5f8276e63655bfc7
Author: sinfallas <sinfallas@gmail.com>
Date:   Thu Jan 1 13:24:48 2015 -0430

    * changelog

commit 3b70807270621b2ed5f8cf1efe5fbb00146ef68f
Author: sinfallas <sinfallas@gmail.com>
Date:   Thu Jan 1 13:23:49 2015 -0430

    * se termina el script de libreoffice
    * se cambia el nombre del script 14.04 a ubuntu

commit 0a6d1305f46451b3ec87147669947a3ca8fad8b0
Author: sinfallas <sinfallas@gmail.com>
Date:   Thu Jan 1 12:04:59 2015 -0430

    * changelog

commit 26c0cb112b44a05770d7d36236e2857b3c00c170
Author: sinfallas <sinfallas@gmail.com>
Date:   Thu Jan 1 12:04:31 2015 -0430

    * se crean los script monitoring y virtualization


12.04.sh
#!/bin/bash
#elaborado por sinfallas
if [ $USER != root ]; then
echo "Error: run as root"
exit 0
fi
nombre=$(whoami)
casa=/home/$nombre
update-pciids
update-usbids
echo -e "net.ipv6.conf.all.disable_ipv6 = 1""\n""net.ipv6.conf.default.disable_ipv6 = 1""\n""net.ipv6.conf.lo.disable_ipv6 = 1" >> /etc/sysctl.d/60-disableipv6.conf
start procps
echo -e 'Section "Device"'"\n"' Identifier "intel"'"\n"' Driver "intel"'"\n"'BusID 	"PCI:0:2:0"'"\n"' Option "AccelMethod" "SNA"'"\n"' Option "SwapbuffersWait" "false"'"\n"' Option "Tiling" "true"'"\n"' Option "BackingStore" "True"'"\n"' Option "XvMC" "on"'"\n"' Option "TripleBuffer" "true"'"\n"' Option "DRI" "true"'"\n"'EndSection' >> /etc/X11/xorg.conf
echo -e "vm.swappiness = 10""\n""vm.dirty_background_ratio = 15""\n""vm.dirty_ratio = 25""\n""vm.vfs_cache_pressure = 50""\n""vm.dirty_writeback_centisecs = 1500""\n""vm.mmap_min_addr = 4096""\n""kernel.shmmni = 4096""\n""kernel.shmall = 2097152""\n""kernel.shmmax = 536870912""\n""kernel.threads-max = 98006""\n""net.ipv6.conf.all.disable_ipv6 = 1""\n""net.ipv6.conf.default.disable_ipv6 = 1""\n""net.ipv6.conf.lo.disable_ipv6 = 1""\n""net.ipv6.conf.all.accept_redirects = 0""\n""net.ipv6.conf.all.accept_source_route = 0""\n""net.ipv4.conf.default.rp_filter=1""\n""net.ipv4.conf.all.rp_filter=1""\n""net.ipv4.conf.all.accept_redirects = 0""\n""net.ipv4.conf.all.send_redirects = 0""\n""net.ipv4.conf.all.accept_source_route = 0""\n""net.ipv4.tcp_timestamps = 1""\n""net.ipv4.tcp_sack = 1""\n""net.ipv4.tcp_window_scaling = 1""\n""net.core.rmem_max = 16777216""\n""net.core.wmem_max = 16777216""\n""net.ipv4.tcp_rmem = 4096 87380 16777216""\n""net.ipv4.tcp_wmem = 4096 65536 16777216""\n""net.ipv4.tcp_no_metrics_save = 1""\n""net.core.netdev_max_backlog = 2500""\n""fs.file-max = 307615""\n""fs.aio-max-nr = 1048576""\n""net.ipv4.tcp_moderate_rcvbuf = 1""\n""net.ipv4.tcp_syncookies=1""\n""# kernel.kptr_restrict = 1""\n""fs.file-max = 307615""\n""fs.aio-max-nr = 1048576" >> /etc/sysctl.conf
sysctl -p
echo "* hard nproc 1000" >> /etc/security/limits.conf
echo -e "/dev/sda {""\n"" lookahead = on""\n"" write_cache = on""\n"" spindown_time = 18""\n""}" >> /etc/hdparm.conf
echo "tmpfs /tmp tmpfs noexec,nosuid,noatime,nodev 0 0" >> /etc/fstab
echo "tmpfs /var/cache/apt/archives tmpfs noatime,nodev 0 0" >> /etc/fstab
echo "tmpfs /var/cache/samba tmpfs noatime,nodev 0 0" >> /etc/fstab
echo "tmpfs /var/log tmpfs noatime,nodev 0 0" >> /etc/fstab
echo "tmpfs /var/spool tmpfs noatime,nodev 0 0" >> /etc/fstab
echo "tmpfs /var/tmp tmpfs noexec,nosuid,noatime,nodev 0 0" >> /etc/fstab
echo "tmpfs /home/$nombre/.cache tmpfs noatime,nodev 0 0" >> /etc/fstab
echo "tmpfs /home/$nombre/.thumbnails tmpfs noatime,nodev 0 0" >> /etc/fstab
echo 'INTEL_BATCH="1"' >> /etc/environment
echo "options rtl8192ce ips=0 swenc=1 fwlps=0 swlps=0" >> /etc/modprobe.d/rtl8192ce.conf
sed -i 's/GRUB_CMDLINE_LINUX_DEFAULT="quiet splash"/GRUB_CMDLINE_LINUX_DEFAULT="elevator=deadline drm.vblankoffdelay=1 i915.i915_enable_fbc=1 i915.i915_enable_rc6=1 i915.lvds_downclock=1 nordrand panic=10 quiet splash"/g' /etc/default/grub
echo -e 'cat << EOF'
'	set superusers="sinfallas"'"\n"'    	password_pbkdf2 sinfallas  grub.pbkdf2.sha512.10000.2FCE22FD822436FBA45D8776887B545011776C04FA66BD8F05A8AB927DA38B707A8286237518BE79DB68348B48B00A2ECC4AA0918EA10CAC11C3DE7CB36FB7EE.3C0AEF540FE5C4C5C284A27C7049D244948D62979657B7D49C347D5F5FE422CC0FA101B5359205011974D587575B6CCF2385C82B98639F7CAD9E44F5B4BB049B'"\n"'EOF' >> /etc/grub.d/00_header 
sed -i 's/  printf "menuentry '${title}' ${CLASS} {\n" "${os}" "${version}"/  printf "menuentry '${title}' ${CLASS} --users sinfallas {\n" "${os}" "${version}"/g' /etc/grub.d/10_linux
update-grub2
sed -i 's/AVAHI_DAEMON_DETECT_LOCAL=1/AVAHI_DAEMON_DETECT_LOCAL=0/g' /etc/default/avahi-daemon
sed -i 's/127.0.0.1	localhost/127.0.0.1	localhost $nombre/g' /etc/hosts
echo "127.0.1.1   geoip.ubuntu.com" >> /etc/hosts
echo -e "[Re-enable hibernate by default]""\n""Identity=unix-user:*""\n""Action=org.freedesktop.upower.hibernate""\n""ResultActive=yes" >> /etc/polkit-1/localauthority/50-local.d/com.ubuntu.desktop.pkla
cd /etc/xdg/autostart/
sed --in-place 's/NoDisplay=true/NoDisplay=false/g' *.desktop
cd $casa

# repositorios
add-apt-repository -y ppa:xorg-edgers/ppa
add-apt-repository -y ppa:glasen/intel-driver
add-apt-repository -y ppa:jacob/virtualisation
add-apt-repository -y ppa:libreoffice/ppa
add-apt-repository -y ppa:ubuntu-wine/ppa
add-apt-repository -y ppa:otto-kesselgulasch/gimp
add-apt-repository -y ppa:pmjdebruijn/darktable-release-plus
add-apt-repository -y ppa:danielrichter2007/grub-customizer
add-apt-repository -y ppa:nilarimogard/webupd8
add-apt-repository -y ppa:tualatrix/ppa
add-apt-repository -y ppa:webupd8team/unstable
add-apt-repository -y ppa:webupd8team/jupiter
add-apt-repository -y ppa:a-v-shkop/chromium
add-apt-repository -y ppa:webupd8team/y-ppa-manager
add-apt-repository -y ppa:ubuntu-mozilla-security/ppa
add-apt-repository -y ppa:thp/gpodder
add-apt-repository -y ppa:diesch/testing
add-apt-repository -y ppa:mumble/release
add-apt-repository -y ppa:richardgv/compton
add-apt-repository -y ppa:xubuntu-dev/xfce-4.10
add-apt-repository -y ppa:rebuntu16/other-stuff 
add-apt-repository -y ppa:kubuntu-ppa/backports
add-apt-repository -y ppa:gnome3-team/gnome3
add-apt-repository -y ppa:ricotz/testing
add-apt-repository -y ppa:alexmurray/indicator-sensors-daily
add-apt-repository -y ppa:team-xbmc/ppa
add-apt-repository -y ppa:webupd8team/tor-browser
add-apt-repository -y ppa:vase/ppa
echo "deb http://download.virtualbox.org/virtualbox/debian precise contrib" >> /etc/apt/sources.list
echo "deb http://dl.google.com/linux/chrome/deb/ stable main" >> /etc/apt/sources.list.d/google.list
wget --output-document=/etc/apt/sources.list.d/medibuntu.list http://www.medibuntu.org/sources.list.d/$(lsb_release -cs).list
wget -q -O - https://dl-ssl.google.com/linux/linux_signing_key.pub | apt-key add - 
wget -q -O - http://download.virtualbox.org/virtualbox/debian/oracle_vbox.asc | apt-key add -
wget -q -O - http://deb.playonlinux.com/public.gpg | apt-key add -
wget http://deb.playonlinux.com/playonlinux_precise.list -O /etc/apt/sources.list.d/playonlinux.list
apt-get -q update
apt-get --allow-unauthenticated -y install medibuntu-keyring
apt-get -q update
apt-get -q -f -y --force-yes dist-upgrade

apt-get -q -f -y --force-yes install build-essential linux-generic-lts-saucy-eol-upgrade linux-firmware linux-firmware-nonfree intel-microcode

apt-get -q -f -y --force-yes install ubuntu-restricted-extras non-free-codecs w64codecs pulseaudio-equalizer preload prelink synaptic ppa-purge y-ppa-manager git xterm pv gcp localepurge bum sysv-rc-conf dkms grub-customizer ssh xserver-xephyr bleachbit macchanger grsync dconf-tools ubuntu-tweak gimp gimp-plugin-registry nautilus-image-converter darktable scribus inkscape calibre vlc audacity acetoneiso isomaster pdfedit gnome-sushi lo-menubar nautilus-dropbox google-chrome-stable chromium-browser-l10n google-talkplugin community-themes libdvdread4 filezilla mumble unetbootin tor-browser libreoffice-style-sifr

apt-get -q -f -y --force-yes install pigz rar p7zip-full unace p7zip-rar sharutils mpack lha arj atool lzop

apt-get -q -f -y --force-yes install gsfonts-other t1-xfree86-nonfree ttf-dustin ttf-f500 ttf-isabella ttf-larabie-deco ttf-larabie-straight ttf-larabie-uncommon ttf-staypuft ttf-summersby ttf-ubuntu-title ttf-xfree86-nonfree xfonts-intl-european gsfonts-x11 ttf-bpg-georgian-fonts ttf-sjfonts
fc-cache -f -v

sed -i 's/\#   ForwardX11 no/ForwardX11 yes/g' /etc/ssh/ssh_config
sed -i 's/Port 22/Port 2222/g' /etc/ssh/sshd_config
sed -i 's/LoginGraceTime 120/LoginGraceTime 20/g' /etc/ssh/sshd_config
sed -i 's/PermitRootLogin yes/PermitRootLogin no/g' /etc/ssh/sshd_config
sed -i 's/\#MaxStartups 10:30:60/MaxStartups 2/g' /etc/ssh/sshd_config
echo -e "MaxAuthTries 2""\n""AllowUsers "$nombre"\n""X11UseLocalhost yes""\n""AddressFamily inet""\n""UseDNS no""\n""ClientAliveInterval 60">> /etc/ssh/sshd_config

#unity
apt-get -q -f -y --force-yes install myunity unsettings compizconfig-settings-manager 
#xfce
apt-get -q -f -y --force-yes install xubuntu-desktop xfce4-composite-editor xfce-theme-manager xfce4-power-manager-plugins
#lxde
apt-get -q -f -y --force-yes install lubuntu-desktop openbox openbox-xdgmenu compton nitrogen tint2 obconf obmenu rxvt-unicode menulibre hal hal-info halevt
echo "@compton --config /home/$nombre/.config/compton.conf -b" >> /etc/xdg/lxsession/Lubuntu/autostart
echo -e "thunar --daemon &""\n""nitrogen --restore &""\n""tint2 &""\n""compton --config /home/$nombre/.config/compton.conf -b &" >> $casa/.config/openbox/autostart
#KDE
apt-get -q -f -y --force-yes install kubuntu-desktop kubuntu-low-fat-settings kubuntu-restricted-extras
#gnome
apt-get -q -f -y --force-yes install gnome-session-fallback gnome gnome-tweak-tool
#enlightenment
apt-get -q -f -y --force-yes install enlightenment

apt-get -q -f -y --force-yes install htop iotop iftop powertop jupiter lm-sensors hddtemp indicator-sensors
echo "coretemp" >> /etc/modules

apt-get -q -f -y --force-yes install playonlinux dosbox freeciv-client-gtk fceux curl pcsxr mupen64plus mupen64plus-extraplugins cutemupen
echo export WINEARCH=win32 >> $casa/.bashrc
mkdir -p $casa/.dosbox
sed -i 's/output=surface/output=overlay/g' $casa/.dosbox/dosbox-0.74.conf
sed -i 's/sensitivity=100/sensitivity=15/g' $casa/.dosbox/dosbox-0.74.conf
sed -i 's/language=/language=spanish-0.74.lang/g' $casa/.dosbox/dosbox-0.74.conf
sed -i 's/memsize=16/memsize=63/g' $casa/.dosbox/dosbox-0.74.conf
sed -i 's/prebuffer=20/prebuffer=50/g' $casa/.dosbox/dosbox-0.74.conf
sed -i 's/oplemu=default/oplemu=fast/g' $casa/.dosbox/dosbox-0.74.conf
sed -i 's/serial1=dummy/serial1=disabled/g' $casa/.dosbox/dosbox-0.74.conf
sed -i 's/serial2=dummy/serial2=disabled/g' $casa/.dosbox/dosbox-0.74.conf
sed -i 's/keyboardlayout=auto/keyboardlayout=es/g' $casa/.dosbox/dosbox-0.74.conf
echo -e "mount c $casa/Dropbox/dosbox""\n""c:""\n""cls" >> $casa/.dosbox/dosbox-0.74.conf

apt-get -q -f -y --force-yes install python-software-properties pkg-config software-properties-common xbmc

apt-get -q -f -y --force-yes install linux-lowlatency virtualbox-4.3 apparmor-utils kvm qemu-kvm-extras qemu qemu-kvm qemu-kvm-spice grub-firmware-qemu qemu-system qemu-user qemuctl vde2 spice-client-gtk python-spice-client-gtk virt-manager virt-goodies virt-viewer bridge-utils

# virtualbox
addgroup vboxusers
adduser $nombre vboxusers
/etc/init.d/vboxdrv setup

# kvm
addgroup kvm
adduser $nombre kvm
addgroup vde2-net
adduser $nombre vde2-net
mkdir -p $casa/isos
mkdir -p $casa/kvm
mkdir -p $casa/kvm/vde
mkdir -p /var/lib/libvirt/images/nas
setcap cap_net_admin=ei /usr/bin/qemu-system-x86_64
echo cap_net_admin $nombre >> /etc/security/capability.conf
sed -i 's/VHOST_NET_ENABLED=0/VHOST_NET_ENABLED=1/g' /etc/default/qemu-kvm
echo "printf '54:52:00:%02X:%02X:%02X\n' $((RANDOM%256)) $((RANDOM%256)) $((RANDOM%256))" >> $kvmdir/kvmgenmac
chmod -v +x $kvmdir/kvmgenmac

xdg-mime default firefox.desktop x-scheme-handler/http
xdg-mime default firefox.desktop x-scheme-handler/https
echo "alias cp='gcp'" >> $casa/.bashrc
echo "alias cp='gcp'" >> /root/.bashrc
echo "alias update='apt-get update && apt-get -y dist-upgrade'" >> /root/.bashrc
echo "alias subir='git add . && git commit -a && git push --all'" >> $casa/.bashrc
sed -i 's/memcached = 0/memcached = 30/g' /etc/preload.conf
sed -i 's/PRELINKING=unknown/PRELINKING=yes/g' /etc/default/prelink
sed -i 's/PRELINK_OPTS=-mR/PRELINK_OPTS=-amR/g' /etc/default/prelink
prelink -amvR
/etc/cron.daily/prelink
/usr/share/doc/libdvdread4/install-css.sh

mv -v /etc/xdg/autostart/zeitgeist-datahub.desktop /etc/xdg/autostart/zeitgeist-datahub.desktop-inactive
mv -v /usr/sbin/modem-manager /usr/sbin/modem-manager-old
mv -v /usr/sbin/aptd /usr/sbin/aptd-old
mv -v /usr/bin/update-notifier /usr/bin/update-notifier-old
mv -v /usr/bin/bluetooth-applet /usr/bin/bluetooth-applet-old
mv -v /usr/bin/blueman-applet /usr/bin/blueman-applet-old
mv -v /usr/lib/indicator-printers/indicator-printers-service /usr/lib/indicator-printers/indicator-printers-service-old
mv -v /usr/lib/unity-lens-music/unity-music-daemon /usr/lib/unity-lens-music/unity-music-daemon.old
mv -v /usr/lib/unity-lens-music/unity-musicstore-daemon /usr/lib/unity-lens-music/unity-musicstore-daemon.old
mv -v /usr/share/oneconf/oneconf-service /usr/share/oneconf/oneconf-service-old 
cd /etc/init.d
chmod -v -x bluetooth
chmod -v -x cups
chmod -v -x gpsd
chmod -v -x modemmanager
chmod -v -x saned
chmod -v -x speech-dispatcher
chmod -v -x brltty
chmod -v -x kdm
update-rc.d -f brltty remove
update-rc.d -f kdm remove
update-rc.d -f saned remove
update-rc.d -f gpsd remove
update-rc.d -f bluetooth remove
update-rc.d -f speech-dispatcher remove
sed -i 's/RUN="no"/RUN="yes"/g' /etc/default/arpon
sed -i 's/# DAEMON_OPTS="-q -f /var/log/arpon/arpon.log -g -s"/DAEMON_OPTS="-q -f /var/log/arpon/arpon.log -g -s"/g' /etc/default/arpon
echo -e "#! /bin/sh""\n""### BEGIN INIT INFO""\n""# Provides:          modulos""\n""# Required-Start:""\n""# Required-Stop:""\n""# Default-Start:     2 3 4 5""\n""# Default-Stop:""\n""# Short-Description: Limpieza""\n""### END INIT INFO""\n""echo always > /sys/kernel/mm/transparent_hugepage/enabled""\n""echo 20000 > /sys/kernel/mm/transparent_hugepage/khugepaged/pages_to_scan""\n""echo 20000 > /sys/kernel/mm/ksm/pages_to_scan""\n""stop avahi-daemon""\n""stop cups""\n""# stop bluetooth""\n""stop kdm""\n""stop qemu-kvm""\n""stop winbind""\n""stop munin-node""\n""stop ssh""\n""stop uml-utilities""\n""stop openvswitch-switch""\n""/etc/init.d/xend stop""\n""/etc/init.d/xcp-squeezed stop""\n""/etc/init.d/xcp-xapi stop""\n""/etc/init.d/xcp-fe stop""\n""/etc/init.d/xcp-v6d stop""\n""/etc/init.d/xcp-networkd stop""\n""rmmod ip6table_filter""\n""rmmod ip6_tables""\n""stop modemmanager""\n""rmmod rfcomm""\n""rmmod bnep""\n""rmmod bluetooth""\n""rmmod msdos""\n""stop libvirt-bin""\n""#ifconfig wlan0 mtu 1492""\n""#ifconfig eth0 mtu 1492""\n""#ifconfig br0 mtu 1492""\n""#ifconfig tap0 mtu 1492""\n""sysctl -p""\n""exit 0" >> /etc/init.d/modulos
chmod -v +x /etc/init.d/modulos
update-rc.d modulos start 96 2 3 4 5 .
cd $casa
apt-get clean
apt-get autoclean
apt-get -y autoremove
clear
echo Finalizado se recomienda reiniciar, elaborado para ubuntu 64 bits por sinfallas.
exit 0
